src/core_modules/capture-core/components/WidgetEventSchedule/hooks/useEventsInOrgUnit.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+import { useMemo, useEffect } from 'react';
+import { useDataQuery } from '@dhis2/app-runtime';
+
+export const useEventsInOrgUnit = (orgUnitId: string, selectedDate: string) => {
+    const { data, error, loading, refetch } = useDataQuery(
+        useMemo(
+            () => ({
+                events: {
+                    resource: 'events',
+                    params: ({ variables: { orgUnitId: id, selectedDate: date } }) => ({
+                        orgUnit: id,
+                        startDate: date,
+                        endDate: date,
+                        skipPaging: true,
+                        status: 'SCHEDULE',
+                        ouMode: 'SELECTED',
+                        fields: 'dueDate',
+                    }),
+                },
+            }),
+            [],
+        ),
+        { lazy: true },
+    );
+
+
+    useEffect(() => {
+        if (orgUnitId && selectedDate) {
+            refetch({ variables: { orgUnitId, selectedDate } });
```

Comment:
It seems like you need to manually select a date for this one to trigger?

Reply:
@JoakimSM I think it's sensible to wait for both orgUnit and selectedDate to be defined before doing the call? What do you suggest here?

Reply:
If the date is auto-calculated (not manually selected/entered) we will also need to show the number of scheduled events. It didn't seem to do the request before I manually selected a date. (correct me if I'm wrong of course)